### PR TITLE
Adapt web socket

### PR DESCRIPTION
**Pull request by juljac56**
Worked on file StudentMeeting.tsx, in the folder Student

Changed made because the websocket communication structure changed in the back (Fast API) : 

- Add an ID to the url of the websocket, so that each student has its own webscoket to communicate with the back
- Add the roomName in the url of the websocket
- Creation of the websocket on launch of the front or on change of the roomName.

See branch websocket_protoco, to understand the change made in the back

### CODE REVIEW

File: box-ui/src/components/ts/Student/StudentMeeting.tsx [hunk 1]
```diff
@@ -65,6 +65,9 @@ const StudentMeeting: FunctionComponent = () => {
     //Animation loading, waiting for new Image
     const [loading, setLoading] = useState<boolean>(false);
 
+    //websocket
+    const [ws1, setWs1] = useState<WebSocket>();
```
Comment (AladdinVR) [inline box-ui/src/components/ts/Student/StudentMeeting.tsx]:
```suggestion
    const [ws, setWs] = useState<WebSocket>();
```
we use only one socket

File: box-ui/src/components/ts/Student/StudentMeeting.tsx [hunk 2]
```diff
@@ -81,29 +84,31 @@ const StudentMeeting: FunctionComponent = () => {
     }, [imgOriginal]);
 
     useEffect(() => {
-        const addressWebsocket = process.env.REACT_APP_WS_ADDRESS;
-        if (addressWebsocket == undefined) {
+        const id = Date.now();
+        const addressWebsocket_general = process.env.REACT_APP_WS_ADDRESS;
+        if (addressWebsocket_general == undefined) {
```
Comment (AladdinVR) [inline box-ui/src/components/ts/Student/StudentMeeting.tsx:90]:
```suggestion
        if (addressWebsocket_general === undefined) {
```

File: box-ui/src/components/ts/Student/StudentMeeting.tsx [hunk 3]
```diff
@@ -81,29 +84,31 @@ const StudentMeeting: FunctionComponent = () => {
     }, [imgOriginal]);
 
     useEffect(() => {
-        const addressWebsocket = process.env.REACT_APP_WS_ADDRESS;
-        if (addressWebsocket == undefined) {
+        const id = Date.now();
```
Comment (djexvr) [inline box-ui/src/components/ts/Student/StudentMeeting.tsx]: It is not a good idea to use the date creation as an identifier even if those are probably unique it is better something made for that. You could either generate it server-side and send it back to the client or use uuid (faster to implement here) which are made for this kind of usage.

File: box-ui/src/components/ts/Student/StudentMeeting.tsx [hunk 4]
```diff
@@ -81,29 +84,31 @@ const StudentMeeting: FunctionComponent = () => {
...
-            }, 1000);
-
             return () => {
-                ws1.close;
-                clearInterval(interval);
+                ws.close;
             };
         }
-    }, [processSelected, information]);
+    }, [ws, processSelected, information]);
```
Comment (djexvr) [inline box-ui/src/components/ts/Student/StudentMeeting.tsx]: You don't information in the dependency here, you don't use it in the callback.

**Pull request merged**